src/batArray.mliv

```diff
@@ -82,8 +82,7 @@ external make : int -> 'a -> 'a array = "caml_make_vect"
     If the value of [x] is a floating-point number, then the maximum
     size is only [Sys.max_array_length / 2].*)
 
-##V<4.2##val make_float : int -> float array
-##V>=4.2##external make_float : int -> float array = "caml_make_float_vect"
+val make_float : int -> float array
```

Comment:
Why did you feel the need for this change? I think that the idea was to follow the interface as present upstream so that the cast `BatArray : module type of Array` would succeed. (I'm not sure it does actually work, though, they may be order-of-fields issues anyway.)